### PR TITLE
replacing local validation with linkml_validate

### PR DESCRIPTION
**Pull request by sierra-moxon**
- improvements to both linkml and linkml-validate that we can take advantage of (updated linkml version in pyproject.toml) including better jsonschema validation
- regenerate python class files as the actions have been failing (protected repo) for awhile.
- run through all the test files to make sure we see no validation failures
  - [x] make some multivalued collections inlined in the schema to match the test data representations
  - [x] move date to datetime
  - [x] fix name_type_name and note_type_name test data
  - [x] remove reference to 'mod.yaml' and replace with the valid allianceMember import
  - [x] `alias` metamodel component to `aliases`
  - [x] fix cases where classes still depended on AuditedObject when they should be represented as Association (OntologyTermClosure)
  - [x] `with` slot is a problem for LinkML - replace with `with_or_from`
  - [x] add more detailed documentation for installing and using pyenv and poetry
  - [x] add directions for running validate on test data without the bespoke validator in this repo (instead use linkml-validate)  
  - [x] allow force push in regenerate-artifacts github action to see if that allows gh-actions user to generate and push artifacts.

### CODE REVIEW

File: test/data/allele_association_ingest_test.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
       "internal": false,
       "obsolete": false,
       "taxon_curie": "NCBITaxon:6239",
-      "date_created": "2015-04-09T10:15:30+00:00",
-      "date_updated": "2022-07-11T13:12:51+00:00",
+      "date_created": "2015-04-09",
```
Comment (markquintontulloch) [inline test/data/allele_association_ingest_test.json:31]: @sierra-moxon Is there a reason for these date changes?  Our Java code expects an OffsetDateTime for all date fields

Comment (sierra-moxon) [reply test/data/allele_association_ingest_test.json:31]: ok awesome - we can change to datetime in linkml (right now these are all "date"s)

**Comment by sierra-moxon** [comment]
@markquintontulloch  - any more comments on this before merge?

File: model/schema/allele.yaml [hunk 2]
```diff
@@ -679,8 +679,9 @@ slots:
     range: string
     required: true
 
-  mutagenesis_methods:
-    alias: ["mutagen"]
+  mutagenesis_method:
+    aliases: ["mutagen"]
+
```
Comment (markquintontulloch) [inline model/schema/allele.yaml:684]: need to remove blank line

File: model/schema/allele.yaml [hunk 3]
```diff
@@ -1056,9 +1059,12 @@ slots:
     range: AlleleNoteSlotAnnotation
 
   allele_note_dtos:
+    is_a: note_dtos
```
Comment (markquintontulloch) [inline model/schema/allele.yaml:1062]: Something for a separate PR, but I think we need to revisit the modelling of this.  We have evidence inherited from the `SlotAnnotation` but also evidence on the `Note` object.  I would have thought that we want to have a similar approach to that of `NameSlotAnnotation` whereby we would have a core NoteSlotAnnotation class the has the `free_text` and `note_type` slots that are currently in `Note`, then the range of `allele_note_dtos` would be `NoteSlotAnnotationDTO`.

File: model/schema/core.yaml [hunk 4]
```diff
@@ -171,12 +171,12 @@ classes:
       A-Team curation UI Controlled Vocabulary Terms Table.
     slots:
       - free_text
-      - note_type
+      - note_type_name
```
Comment (markquintontulloch) [inline model/schema/core.yaml:174]: This should still be `note_type` with a range of `VocabularyTerm`.  The `NoteDTO` class  uses `note_type_name`

**Review by markquintontulloch — CHANGES_REQUESTED** [review]

File: model/schema/allele.yaml [hunk 5]
```diff
@@ -123,7 +123,7 @@ classes:
   GenerationMethod:
     is_a: AuditedObject
     slots:
-      - mutagenesis_methods
+      - mutagenesis_method_names
```
[the remark this answers is not in the document]
Comment (sierra-moxon) [reply model/schema/allele.yaml:126]: @markquintontulloch - there is no `mutagenesis_methods` in the schema...should it be `mutagenesis_method` or `mutagenesis_method_names` or should `mutagenesis_methods` be added?

File: model/schema/allele.yaml (hunk 3 above)
Comment (sierra-moxon) [reply model/schema/allele.yaml:1062]: Yes I agree - it is inconsistent.

File: model/schema/allele.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (markquintontulloch) [reply model/schema/allele.yaml]: @sierra-moxon - it is `mutagenesis_methods` on the main branch

File: model/schema/bulkload.yaml [hunk 6]
```diff
@@ -135,7 +135,7 @@ slots:
     description: >-
       Indicates when the file was last loaded
     domain: BulkLoadFile
-    range: date  
+    range: datetime
```
Comment (chris-grove) [inline model/schema/bulkload.yaml:138]: @sierra-moxon 
Is this intended to be a universal change (across the entire model) from a range of 'date' to 'datetime'? If so, there are two other locations still using 'date':
https://github.com/alliance-genome/agr_curation_schema/blob/main/model/schema/resource.yaml#L93
https://github.com/alliance-genome/agr_curation_schema/blob/main/model/schema/curationReport.yaml#L106
If those are fine to stay as 'date' and not 'datetime' then you can ignore.

Comment (sierra-moxon) [reply model/schema/bulkload.yaml:138]: No definitely not, the new validation just does a better job of checking that the sample data we provide matches the schema.  So in our test data, we use datetime values (which is required by the java app layer, I think, right?), and our old validator wasn't checking that date was different from datetime.  Now it is so I fixed our schema to match the data.  

Other downstream apps (blue team work?) might require simple dates (we need tests to cover more of our schema), so this is a change that needs to happen on a case-by-case basis.

File: README.md [hunk 7]
```diff
@@ -175,25 +175,57 @@ git clone https://github.com/alliance-genome/agr_curation_schema
...
+python environment).  Then run the validator:
 ```bash
-python util/validate_agr_schema.py -i path_to_the_json_data_to_validate
+poetry run linkml-validate -C Ingest -s model/schema/allianceModel.yaml -s model/schema/allianceModel.yaml [path/to/your/submission/file.json]
 ```
 
 example command using test data (replace test/allele_ingest_test.json with the path to your submission.json):
 ```bash
-python util/validate_agr_schema.py -i test/allele_ingest_test.json
+run linkml-validate -C Ingest -s model/schema/allianceModel.yaml -s model/schema/allianceModel.yaml test/allele_ingest_test.json
```
Comment (chris-grove) [inline README.md:186]: Note: I need to include 'poetry' here otherwise my command line complains of no 'run' command

**Review by markquintontulloch — APPROVED** [review]

**Comment by chris-grove** [comment]
@sierra-moxon 
Everything looks good! My 'make test' worked fine. On every test run I was getting the message:
`.../agr_curation_schema/venv/lib/python3.7/site-packages/rdflib_jsonld/__init__.py:12: DeprecationWarning: The rdflib-jsonld package has been integrated into rdflib as of rdflib==6.0.0.  Please remove rdflib-jsonld from your project's dependencies.`
Other than that everything seems fine. I did add the 'poetry' command to the README.md file where I commented.

**Comment by sierra-moxon** [comment]
> @sierra-moxon Everything looks good! My 'make test' worked fine. On every test run I was getting the message: `.../agr_curation_schema/venv/lib/python3.7/site-packages/rdflib_jsonld/__init__.py:12: DeprecationWarning: The rdflib-jsonld package has been integrated into rdflib as of rdflib==6.0.0. Please remove rdflib-jsonld from your project's dependencies.` Other than that everything seems fine. I did add the 'poetry' command to the README.md file where I commented.

thanks so much for this @chris-grove - I know it probably was a lot of work on top of your already big list!  :)

**Pull request merged**